### PR TITLE
test(js): Convert SentryApplicationDetails to RTL

### PR DESCRIPTION
**Pull request by evanpurkhiser**

### CODE REVIEW

File: static/app/views/settings/organizationDeveloperSettings/sentryApplicationDetails.spec.jsx [hunk 1]
```diff
@@ -15,8 +14,6 @@ describe('Sentry Application Details', function () {
   let createAppRequest;
   let editAppRequest;
 
-  const verifyInstallToggle = 'Switch[name="verifyInstall"]';
```
Comment (priscilawebdev) [inline static/app/views/settings/organizationDeveloperSettings/sentryApplicationDetails.spec.jsx:18]: lint is saying that the `wrapper` is defined but not used. shall we remove it?

File: static/app/views/settings/organizationDeveloperSettings/sentryApplicationDetails.spec.jsx [hunk 2]
```diff
@@ -154,38 +177,50 @@ describe('Sentry Application Details', function () {
...
 
-    it('shows application data', function () {
-      // data should be filled out
-      expect(wrapper.find('PermissionsObserver').prop('scopes')).toEqual([
-        'project:read',
-      ]);
+    it('shows application data', async function () {
+      renderComponent();
+
+      selectEvent.openMenu(screen.getByRole('textbox', {name: 'Project'}));
```
Comment (priscilawebdev) [inline static/app/views/settings/organizationDeveloperSettings/sentryApplicationDetails.spec.jsx:202]: or shall we add an `await` here?

Comment (evanpurkhiser) [reply static/app/views/settings/organizationDeveloperSettings/sentryApplicationDetails.spec.jsx:201]: `openMenu` isn't async

**Review by priscilawebdev — APPROVED** [review]
🛳🇮🇹

**Pull request merged**